lib/contract.js

```diff
@@ -46,8 +46,10 @@ var contract = function (address, desc) {
                 call: function (extra) {
                     extra = extra || {};
                     extra.to = address;
-                    extra.data = parsed;
-                    return web3.eth.call(extra).then(onSuccess);
+                    return abi.methodSignature(desc, method.name).then(function (signature) {
+                        extra.data = signature.slice(0, 10) + parsed;
+                        return web3.eth.call(extra).then(onSuccess);
+                    });
```

Comment:
it can be improved by caching method signature for next calls, or by getting it earlier